atoma-service/src/handlers/chat_completions.rs

```diff
@@ -826,11 +826,36 @@ async fn handle_streaming_response(
         })?;
 
     if !response.status().is_success() {
-        let error = response
-            .status()
-            .canonical_reason()
-            .unwrap_or("Unknown error");
-        handle_status_code_error(response.status(), &endpoint, error)?;
+        let status = response.status();
+        let bytes = response
+            .bytes()
+            .await
+            .map_err(|e| AtomaServiceError::InternalError {
+                message: format!("Failed to read response body: {}", e),
+                endpoint: endpoint.to_string(),
+            })?;
+        // Try to parse the error message from the response body
+        let error_message = serde_json::from_slice::<Value>(&bytes).map_or_else(
```

Comment:
Is this really necessary ? Isn't there a way to plug the vllm/sglang logs into loki directly ?